Automattic/jetpack
# Add conditional recommendations to the Jetpack assistant

#### Changes proposed in this Pull Request:

* This PR adds logic to show recommendations in the Jetpack assistant based on actions that occur on a site.
* In this PR, publishing a new post will enable a recommendation for Publicize to show in the assistant interface.

#### Jetpack product discussion


p1HpG7-dON-p2

#### Does this pull request change what data or activity we track or use?


* Starts keeping track of which recommendations have been viewed as part of the `recommendations_data` option.
* Adds a new `recommendations_conditional` option, which saves the conditional recommendations that are enabled for the site.
* This PR adds two new steps to the recommendations flow (Publicize and VideoPress). These steps do not add any new analytics events, but they use current analytics events to record activity on the related step screens and summary screen, passing the step/feature slug as a property to the event. The existing analytics events used for these new steps are:
```
// Individual Steps
jetpack_recommended_feature_learn_more_click
jetpack_recommended_feature_enable_click
jetpack_recommended_feature_decide_later_click

// Summary screen
jetpack_recommendations_summary_configure_click
jetpack_recommendations_summary_learn_more_click
jetpack_recommendations_summary_enable_click
```

#### Testing instructions:






To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`

If your test site already has the Publicize module enabled, you will not receive the new recommendation as outlined in the test plan below. Be sure to disable the Publicize module before testing.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
![Screen Shot 2022-03-21 at 2 27 58 PM](https://user-images.githubusercontent.com/18016357/159340092-2adb1a7e-7834-4986-b965-33519fa1ee11.png)


* Now, go create and publish a new post (or switch a post to draft and back to publish)
* After the post is published, navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
![Screen Shot 2022-03-11 at 4 00 23 PM](https://user-images.githubusercontent.com/18016357/158703217-3b7bb9bb-a1e9-4b6a-814e-d81e167094f1.png)

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation to enable the Publicize feature. There should be a green "New" badge that shows on this step next to the progress bar.
![Screen Shot 2022-03-11 at 4 58 50 PM](https://user-images.githubusercontent.com/18016357/158703637-a1d7d220-8820-4ea4-b8a0-1c0e60ce7ab9.png)

* Proceed through the rest of the flow to the summary screen. On the summary screen, you should see the "Publicize" feature listed and a green "New" badge next to it in the list.
![Screen Shot 2022-03-11 at 4 59 06 PM](https://user-images.githubusercontent.com/18016357/158703673-acf6465e-a44a-44cc-89d3-e2bdb945eeba.png)

* If you reload the page on the summary screen, the green "New" badge next to Publicize in the summary feature list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Test that the buttons to enable Publicize from both the individual recommendation screen and the summary screen work and enable the correct module.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570
